cdap-explore/src/main/java/co/cask/cdap/explore/service/ExploreServiceUtils.java

```diff
@@ -496,4 +513,14 @@ private static File updateHiveConfFile(File confFile, File tempDir) {
     }
     return newHiveConfFile;
   }
+
+  public static boolean isSparkAvailable() {
+    return System.getenv(Constants.SPARK_HOME) != null;
+  }
+
+  public static boolean isSparkEngine(HiveConf hiveConf) {
+    // We don't support setting engine through session configuration now
+    String engine = hiveConf.get("hive.execution.engine");
+    return engine != null && engine.equalsIgnoreCase("spark");
```

Comment:
You can just do `"spark".equalsIgnoreCase(engine)`, then you don't have to check for `null`.